### PR TITLE
support proxy_connect_address and proxy_connect_bind directives

### PR DESCRIPTION
**Pull request by skyever**
add proxy_connect_address and proxy_connect_bind directives
 * proxy_connect_address: Specifiy an IP address of the proxied server.
 * proxy_connect_bind: Makes outgoing connections to a proxied server
   originate from the specified local IP address with an optional port.

Signed-off-by: fengshuaitao <fengshuaitao@bytedance.com>

### CODE REVIEW

File: t/http_proxy_connect.t [hunk 1]
```diff
@@ -39,23 +39,40 @@ http {
...
+        rewrite_by_lua_block {
+            if ngx.var.host == "address.com" then
+                ngx.var.proxy_remote_address = "127.0.0.1:8082"
+            end
+
+            if ngx.var.host == "bind.com" then
+                ngx.var.proxy_remote_address = "127.0.0.1:8083"
+                ngx.var.proxy_local_address = "127.0.0.3"
+            end
+        }
```
Comment (chobits) [inline t/http_proxy_connect.t]: should use if-set directive, which needs less dependence.

```
if ($host = "address.com") {
      set $proxy_remote_address "127.0.0.1:8082";
}
...
```

**Comment by skyever** [comment]
@chobits Thanks for ur reviewing, patch updated, pls check

File: README.md [hunk 2]
```diff
@@ -135,6 +135,35 @@ Sets a timeout for transmitting a request to the proxied server.
...
 If the proxied server does not receive anything within this time, the connection is closed.
 
+proxy_connect_address
+---------------------------
+
+Syntax: **proxy_connect_address `address [transparent] | off`**
+Default: `none`
+Context: `server`
+
+Specifiy an IP address of the proxied server. The address can contain variables. The special value is equal to none,
```
Comment (chobits) [inline README.md]: `The special value is equal to none,`

->

`The default value is none,`

Comment (skyever) [reply README.md]: uh, it's a typo.

**Comment by chobits** [comment]
thanks merged!

**Pull request merged**